src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -236,6 +241,19 @@ export const useSearch = (services: DiscoverViewServices) => {
           elapsedMs,
         },
       });
+
+      // Add tracking for datasource type and query if the query is not same as last query
+      const query = searchSource.getField('query');
```

Comment:
can't remember if discover creates a new search source and it doesn't have a parent but we could also check if searchSource.parent.getField('query') . but then again i haven't checked so it parent might not ever be set for discover